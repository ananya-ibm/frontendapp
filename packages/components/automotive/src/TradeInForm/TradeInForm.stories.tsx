/*
Licensed Materials - Property of IBM
694906H
(c) Copyright IBM Corp.  2020 All Rights Reserved

US Government Users Restricted Rights - Use, duplication or disclosure restricted
by GSA ADP Schedule Contract with IBM Corp.
*/

import React from 'react';
import { TradeInForm } from './TradeInForm';

export default {
  title: 'Components/Automotive/TradeInForm',
  component: TradeInForm
};

const storyProps = {
  carCondition: {
    label: 'Car Condition',
    options: ['Excellent', 'Good', 'Fair', 'Poor']
  },
  formProps: {
    onSubmit: () => {}
  }
};

const initialValues = {
  registration: 'ABC123',
  mileage: '10000'
};

export const normal = args => <TradeInForm {...args} />;
normal.args = storyProps;

export const prePopulated = args => (
  <TradeInForm {...args} formProps={{ ...storyProps.formProps, initialValues }} />
);
prePopulated.args = storyProps;
