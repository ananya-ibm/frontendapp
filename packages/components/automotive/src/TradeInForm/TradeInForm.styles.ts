/*
Licensed Materials - Property of IBM
694906H
(c) Copyright IBM Corp.  2020 All Rights Reserved

US Government Users Restricted Rights - Use, duplication or disclosure restricted
by GSA ADP Schedule Contract with IBM Corp.
*/

import styled from 'styled-components';

export const TradeInForm = styled('div')`
  & .submit {
    display: block;
    margin-top: ${props => props.theme.spacing.stack.s7};
  }
`;
