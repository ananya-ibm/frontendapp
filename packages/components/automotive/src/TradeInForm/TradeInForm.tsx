/*
Licensed Materials - Property of IBM
694906H
(c) Copyright IBM Corp.  2020 All Rights Reserved

US Government Users Restricted Rights - Use, duplication or disclosure restricted
by GSA ADP Schedule Contract with IBM Corp.
*/

import React from 'react';
import { yupResolver } from '@hookform/resolvers/yup';
import { Button } from '@exo/frontend-components-base';
import * as yup from 'yup';
import { BaseForm, Field, TextInput, Dropdown } from '@exo/frontend-components-forms';
import { useForm } from 'react-hook-form';
import * as S from './TradeInForm.styles';

export const TradeInForm = ({ onClickTerms, carCondition, formProps }: Props) => {
  const schema = yup.object().shape({
    registration: yup.string().required(),
    mileage: yup
      .number()
      .required()
      .positive(),
    condition: yup.string().required()
  });

  const { register, handleSubmit, formState, control, reset } = useForm({
    mode: 'onChange',
    resolver: yupResolver(schema)
  });

  return (
    <S.TradeInForm>
      <BaseForm
        onSubmit={formProps?.onSubmit}
        data={formProps?.initialValues}
        renderFooter={() => <Button type="submit" label="Search Vehicle" />}
        form={{ handleSubmit, formState, reset }}
      >
        <Field>
          <TextInput
            id="registration"
            {...register('registration', { required: true, minLength: 6 })}
            isRequired={true}
            control={control}
            value={formProps?.initialValues?.registration}
            errorText={formState.errors.registration && 'Incorrect registration number'}
            labelText={'Registration Number'}
            placeholderText={'Enter your registration number'}
          />
        </Field>

        <Field>
          <TextInput
            id="mileage"
            {...register('mileage', { required: true })}
            isRequired={true}
            control={control}
            value={formProps?.initialValues?.mileage}
            errorText={formState.errors.mileage && 'Incorrect mileage'}
            labelText={'Car Mileage'}
            placeholderText={"Enter your car's mileage"}
          />
        </Field>

        <Field>
          <Dropdown
            id="condition"
            {...register('condition', { required: true })}
            isRequired={true}
            control={control}
            value={formProps?.initialValues?.condition}
            errorText={formState.errors.condition && 'Incorrect condition'}
            labelText={'Car Condition'}
            placeholderText={"Enter your car's condition..."}
            items={carCondition.options.map(o => ({ name: o, value: o }))}
          />
        </Field>

        <Button variant="link" label="Terms and conditions" onClick={onClickTerms} />
      </BaseForm>
    </S.TradeInForm>
  );
};

type Props = {
  onClickTerms?: () => void;
  carCondition: {
    label: string;
    options: string[];
  };
  formProps: {
    onSubmit: (a: any) => void;
    initialValues?: {
      mileage?: string | number;
      registration?: string;
      condition?: string;
    };
  };
};
