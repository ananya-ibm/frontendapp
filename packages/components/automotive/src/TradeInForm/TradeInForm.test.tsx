/*
Licensed Materials - Property of IBM
694906H
(c) Copyright IBM Corp.  2020 All Rights Reserved

US Government Users Restricted Rights - Use, duplication or disclosure restricted
by GSA ADP Schedule Contract with IBM Corp.
*/

import React from 'react';
import { renderWithTheme as render } from '@testUtils';
import { TradeInForm } from './TradeInForm';

const testProps = {
  carCondition: {
    label: 'Car Condition',
    options: ['Excellent', 'Good', 'Fair', 'Poor']
  },
  formProps: {
    onSubmit: () => {}
  }
};

describe('<TradeInForm /> component', () => {
  test('its snapshot matches', () => {
    const { container } = render(<TradeInForm {...testProps} />);

    expect(container).toBeDefined();
    expect(container.firstChild).toMatchSnapshot();
  });
});
