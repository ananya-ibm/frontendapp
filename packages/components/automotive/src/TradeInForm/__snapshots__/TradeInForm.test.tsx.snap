// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TradeInForm /> component its snapshot matches 1`] = `
.c2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: none;
  border: none;
  color: #0f62fe;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font: bold 0.875rem/1.125005rem 'IBM Plex Sans',Arial,sans-serif;
  -webkit-letter-spacing: 0.16px;
  -moz-letter-spacing: 0.16px;
  -ms-letter-spacing: 0.16px;
  letter-spacing: 0.16px;
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c2:hover {
  color: #0043ce;
}

.c2 svg {
  color: #161616;
}

.c4 {
  font-size: 105%;
  font-weight: 700;
  line-height: 0;
  white-space: nowrap;
}

.c4 svg {
  max-height: 100%;
}

.c1 {
  margin-bottom: 1.5rem;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding-top: 1.5rem;
}

.c0 .submit {
  display: block;
  margin-top: 2rem;
}

<div
  class="c0"
>
  <form>
    <div
      class="FormBodystyles__FormBody-sc-12wq4uu-0"
    >
      <div
        class="c1"
      >
        <div
          class="cds--form-item cds--text-input-wrapper"
        >
          <label
            class="cds--label"
            for="registration"
          >
            Registration Number
          </label>
          <div
            class="cds--text-input__field-outer-wrapper"
          >
            <div
              class="cds--text-input__field-wrapper"
            >
              <input
                class="cds--text-input cds--text__input cds--text-input--md"
                id="registration"
                name="registration"
                placeholder="Enter your registration number"
                required=""
                title="Enter your registration number"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="c1"
      >
        <div
          class="cds--form-item cds--text-input-wrapper"
        >
          <label
            class="cds--label"
            for="mileage"
          >
            Car Mileage
          </label>
          <div
            class="cds--text-input__field-outer-wrapper"
          >
            <div
              class="cds--text-input__field-wrapper"
            >
              <input
                class="cds--text-input cds--text__input cds--text-input--md"
                id="mileage"
                name="mileage"
                placeholder="Enter your car's mileage"
                required=""
                title="Enter your car's mileage"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="c1"
      >
        <div
          class="cds--dropdown__wrapper cds--list-box__wrapper"
        >
          <label
            class="cds--label"
            for="downshift-0-toggle-button"
            id="downshift-0-label"
          >
            Car Condition
          </label>
          <div
            class="cds--dropdown cds--list-box"
            id="condition"
          >
            <button
              aria-disabled="false"
              aria-expanded="false"
              aria-haspopup="listbox"
              aria-labelledby="downshift-0-label downshift-0-toggle-button"
              class="cds--list-box__field"
              id="downshift-0-toggle-button"
              title="Enter your car's condition..."
              type="button"
            >
              <span
                class="cds--list-box__label"
              >
                Enter your car's condition...
              </span>
              <div
                class="cds--list-box__menu-icon"
              >
                <svg
                  aria-label="Open menu"
                  fill="currentColor"
                  focusable="false"
                  height="16"
                  name="chevron--down"
                  preserveAspectRatio="xMidYMid meet"
                  role="img"
                  viewBox="0 0 16 16"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M8 11L3 6 3.7 5.3 8 9.6 12.3 5.3 13 6z"
                  />
                  <title>
                    Open menu
                  </title>
                </svg>
              </div>
            </button>
            <div
              aria-labelledby="downshift-0-label"
              class="cds--list-box__menu"
              id="downshift-0-menu"
              role="listbox"
              tabindex="-1"
            />
          </div>
        </div>
      </div>
      <a
        class="c2"
      >
        Terms and conditions
      </a>
    </div>
    <div
      class="c3"
    >
      <button
        class="c4 cds--btn cds--btn--primary"
        tabindex="0"
        type="submit"
      >
        Search Vehicle
         
      </button>
    </div>
  </form>
</div>
`;
